syntax = "proto3";

package eu.surething_project.core;

option java_multiple_files = true;
option java_package = "eu.surething_project.core.grpc";
option java_outer_classname = "LocCertProto";

import "google/protobuf/any.proto";
import "loc_time.proto";
import "signature.proto";

// Location Certificate generated by the verifier for a successful verification of a prover's location claim
message LocationVerification {
	// unique ID for this verification
	string verificationId = 1;
	// verifier unique id	
	string verifierId = 2;
	// the original location claim of the prover
	string claimId = 3;
	// list of location endorsements by witnesses
	repeated string endorsementIds = 4;
	// time of verification - in one of alternative time schemas
	Time time = 5;
	// type of the evidence
	string evidenceType = 6;
	// verification evidence
	google.protobuf.Any evidence = 7;
}

// A location certificate is a verified location claim, optionally supported by endorsements
message LocationCertificate {
	LocationVerification verification = 1;
	Signature verifierSignature = 2;
}
